scripts/check_edit_location.py

```diff
@@ -0,0 +1,211 @@
+            layer_query_string = layer_query_string + f"{editor_field} = '{worker}'"
+    else:
+        logger.info("Please pass at least one worker user_id")
+        sys.exit()
+    # These are the features whose corresponding editors we will check
+    sr = {'wkid': 3857, 'latestWkid': 3857}
+    features_df = layer.query(where=layer_query_string, out_sr=sr, return_all_records=True, as_df=True)
+    if len(features_df.index) == 0:
+        logger.info("No features found to check. Please check the user_id's that you have passed")
+        sys.exit(0)
+    
+    # buffer features to use as geometry filter
+    features_df["BUFFERED"] = features_df["SHAPE"].geom.buffer(dist_tolerance + min_accuracy)
+    features_df.spatial.set_geometry("BUFFERED")
+    features_df.spatial.sr = sr
+    
+    # Set field names
+    accuracy_field = "horizontal_accuracy"
+    creator_field = "created_user"
+    timestamp_field = "location_timestamp"
+    
+    # Find invalid features
+    invalid_features = []
+    logger.info("Finding invalid features")
+    for index, row in features_df.iterrows():
+        # The date to check
+        try:
+            # date field may not be populated
+            if not pandas.isnull(row[field_name]):
+                date_to_check = row[field_name]
+            else:
+                continue
+        except Exception as e:
+            logger.info("Check that the exact field name exists in the feature layer")
+            logger.info(e)
+            sys.exit(0)
+        
+        # Add/Subtract some minutes to give a little leeway
+        start_date = date_to_check - datetime.timedelta(minutes=time_tolerance)
+        end_date = date_to_check + datetime.timedelta(minutes=time_tolerance)
+        
+        # Check there are actually tracks in your LTS in that time period. Otherwise, go to next feature
+        check_track_query = f"{timestamp_field} < '{end_date.strftime('%Y-%m-%d %H:%M:%S')}'"
+        check_tracks = tracks_layer.query(where=check_track_query, return_count_only=True)
+        if check_tracks == 0:
+            logger.info("For this feature, no tracks exist for the time period in your LTS. Ensure that tracks have been retained for the time period you're verifying")
+            continue
+        
+        # Make a query string to select location by the worker during the time period
+        loc_query_string = f"{creator_field} = '{row[editor_field]}' " \
+            f"AND {timestamp_field} >= '{start_date.strftime('%Y-%m-%d %H:%M:%S')}' " \
+            f"AND {timestamp_field} <= '{end_date.strftime('%Y-%m-%d %H:%M:%S')}' " \
+            f"AND {accuracy_field} <= {min_accuracy}" \
+
+        # Generate geometry filter, query the feature layer
+        geom_filter = arcgis.geometry.filters.intersects(row['BUFFERED'], sr=sr)
+        tracks_within_buffer = tracks_layer.query(where=loc_query_string, geometry_filter=geom_filter, out_sr=sr).features
```

Comment:
I think if you did this, you could remove the `check_track_query` above

Reply:
You'd still want to make sure that tracks have been retained for that time period - this is basically a way for me to check retention period without having to do all that string conversion from how it's stored in the admin properties. Otherwise, you could return 0 tracks and report it as an invalid feature, when you don't really know if that's true. Agree w changing to `return_count_only` though